src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_field_tuples.jl

```diff
@@ -27,8 +27,8 @@ end
 function hydrostatic_tendency_fields(velocities, free_surface::SplitExplicitFreeSurface, grid, tracer_names)
     u = XFaceField(grid)
     v = YFaceField(grid)
-    U = deepcopy(free_surface.barotropic_velocities.U)
-    V = deepcopy(free_surface.barotropic_velocities.V)
+    U = similar(free_surface.barotropic_velocities.U)
+    V = similar(free_surface.barotropic_velocities.V)
```

Comment:
This is unrelated right? The difference is that “similar” doesn’t copy boundary conditions.

Reply:
Right, this was a leftover from the Metal branch (that I stripped down for this PR). Apparently, there is a bug with `deepcopy` and Metal OffsetArrays:
```julia
julia> a = MtlArray(zeros(Float32, 10));

julia> a = OffsetArray(a, -1);

julia> b = deepcopy(a);

julia> fill!(b, 1);

julia> a
10-element OffsetArray(::MtlVector{Float32, Metal.PrivateStorage}, 0:9) with eltype Float32 with indices 0:9:
 1.0
 1.0
 1.0
 1.0
 1.0
 1.0
 1.0
 1.0
 1.0
 1.0
```
I wanted to raise an issue in OffsetArrays.jl but I haven't done it yet (there is some type-piracy of Base going on in that package).

Reply:
with similar, on the other hand, there is no issue:
```julia
julia> a = MtlArray(zeros(Float32, 4));

julia> a = OffsetArray(a, -1);

julia> b = similar(a);

julia> a
4-element OffsetArray(::MtlVector{Float32, Metal.PrivateStorage}, 0:3) with eltype Float32 with indices 0:3:
 0.0
 0.0
 0.0
 0.0

```

Reply:
weird